lib/pbench/server/auth/__init__.py

```diff
@@ -115,18 +115,30 @@ def set_well_known_endpoints(self):
             )
             raise
 
-    def get_oidc_public_key(self, token: str):
+    def _get_oidc_public_key(self) -> str:
+        """Returns the public key of the current OIDC client.
+
+        Returns:
+            public key string
+
+        Raises:
+            KeyError: if public key not found in response payload
+            requests.exceptions: Requests session Exceptions
+        """
+
+        realm_public_key_uri = f"{self.server_url}/realms/{self.realm_name}"
+        response_json = self._get(realm_public_key_uri).json()
+        public_key = response_json["public_key"]
+        return (
+            "-----BEGIN PUBLIC KEY-----\n" + public_key + "\n-----END PUBLIC KEY-----"
+        )
+
+    def get_oidc_public_key(self):
```

Comment:
A simple change to avoid the `_get_oidc_public_key` call in the constructor would be to defer that to here:

```python
    if not self.public_key:
        self.public_key = self._get_oidc_public_key()
    return self.public_key
```

Reply:
This is a superb idea, although it will require initializing `public_key` to `None` (or similar) in `__init__()` (at line 64)...otherwise, we'll get an `AttributeError` in the test here.

Reply:
I think that’s a great suggestion, I will update the code.